lib/rspec/mocks/message_expectation.rb

```diff
@@ -354,6 +354,25 @@ def to_s
       end
       alias inspect to_s
 
+      private
+
+      if RSpec::Support::RubyFeatures.kw_args_supported?
+        def __call_original(original, *args, &block)
+          if RSpec::Support::MethodSignature.new(original).has_kw_args_in?(args)
+            binding.eval(<<-CODE, __FILE__, __LINE__)
```

Comment:
Still not super fan of using this. But we already had a discussion, and it seems the best solution. :)